### PR TITLE
Add pretty printer for blocks

### PR DESCRIPTION
**Pull request by KentShikama**
## Overview
Provide a brief description of what this PR does, and why it's needed.

Example output

```
Block #2 (9e99bd8eb4...) -- Creator ID 56616c6964... -- M Parent Hash 7ff1b1ef7c... -- Contents [Produce 2]
```

### Does this PR relate to an RChain JIRA issue? 
If applicable, add link to corresponding JIRA issue.

https://rchain.atlassian.net/browse/RHOL-278

### CODE REVIEW

File: casper/src/main/scala/coop/rchain/casper/PrettyPrinter.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import coop.rchain.casper.protocol.Resource.ResourceClass.{
+  ConsumeResource,
+  ProduceResource,
+  StoppedResource
+}
+import coop.rchain.casper.protocol.{BlockMessage, Resource, Stopped}
+import scalapb.GeneratedMessage
+import coop.rchain.crypto.codec._
+
+class PrettyPrinter {
```
Comment (goral09) [inline casper/src/main/scala/coop/rchain/casper/PrettyPrinter.scala]: Why is this a `class` if it only contains functions? Maybe `object` will be enough?

File: casper/src/main/scala/coop/rchain/casper/PrettyPrinter.scala [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            s"-- M Parent Hash ${buildString(mainParent)} " +
+            s"-- Contents ${postState.resources.toList
+              .map {
+                case Resource(resourceClass) =>
+                  resourceClass match {
+                    case ProduceResource(produce) => s"Produce ${produce.id}"
+                    case ConsumeResource(consume) => s"Consume ${consume.id}"
+                    case StoppedResource(_)       => ""
+                  }
+              }
```
Comment (goral09) [inline casper/src/main/scala/coop/rchain/casper/PrettyPrinter.scala]: I'm fine with string interpolation but I would rather move bigger chunks of code outside .

File: casper/src/main/scala/coop/rchain/casper/PrettyPrinter.scala (hunk 1 above)
Comment (KentShikama) [reply casper/src/main/scala/coop/rchain/casper/PrettyPrinter.scala]: Done

File: casper/src/main/scala/coop/rchain/casper/PrettyPrinter.scala (hunk 2 above)
Comment (KentShikama) [reply casper/src/main/scala/coop/rchain/casper/PrettyPrinter.scala]: Done

**Comment by KentShikama** [comment]
@goral09 @birchmd Ready for another review

**Pull request merged**